backend/src/datasource.js

```diff
@@ -0,0 +1,67 @@
+import { DataSource } from "apollo-datasource";
+import crypto from "crypto";
+
+export class Post {
+    constructor(data) {
+        this.id = crypto.randomBytes(16).toString("hex");
+        this.voters = new Map();
+        Object.assign(this, data);
```

Comment:
I really, _really_ dislike this way of declaring constructors and initializing objects.
I can't really fault you for this because @roschaefer used and showed this off in the lecture, but this way you'll loose all the knowledge about which properties `Post` has and can be setup with..

Reply:
That's a legitimate interest, if your build tools support it you could either use [instance fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields) or you could just do:
```js
const { some, white, listed, arguments } = data
Object.assign(this, { some, white, listed, arguments }); 
```